.github/workflows/publish.yml

```diff
@@ -45,6 +49,17 @@ jobs:
         with:
           repository-url: https://test.pypi.org/legacy/
 
+      # Remove the PR label which triggered this release,
+      # but only if the release failed. Presumably, in this case we might want
+      # to fix whatever caused the failure and re-trigger the test release.
+      # If the release succeeded, re-triggering the release would fail anyway,
+      # unless the version would be bumped again.
+      - name: Remove test-release label
+        if: failure()
```

Comment:
https://stackoverflow.com/a/58859404/3682277

Reply:
We could also decide to "always" remove the label, regardless of whether the publish was successfull or not. Or we can get rid of this altogether if this seems overly complicated. But maybe I'd do that as a separate PR so that we have a record of a working version in this repo. Took me a while to figure this one out, as is evident from the commits. :-D

Reply:
I like it, let's see how it works in the future and then we can improve. It's definitely good to keep it in history of this repo in case we need it.